frontend/src/concepts/metrics/kserve/utils.ts

```diff
@@ -0,0 +1,27 @@
+import { ConfigMapKind } from '~/k8sTypes';
+import {
+  KserveMetricsConfigMapKind,
+  KserveMetricsDataObject,
+} from '~/concepts/metrics/kserve/types';
+
+export const isKserveMetricsConfigMapKind = (
+  configMapKind: ConfigMapKind,
+): configMapKind is KserveMetricsConfigMapKind =>
+  (configMapKind as KserveMetricsConfigMapKind).data.supported === 'true' ||
+  (configMapKind as KserveMetricsConfigMapKind).data.supported === 'false';
```

Comment:
Not sure exactly what kind of values will be passed into this, but `data` is possibly `undefined` for a config map and there's no defensive checks here.

Reply:
If you can try to do this without using the keyword `as` that would be good as we're attempting to remove the usage of casting. You can do checks like `'data' in configMapKind`

Reply:
Ahh didn't see the followup, I've removed the `as` - is this up to standard now?